dask/dataframe/io/parquet.py

```diff
@@ -229,8 +241,38 @@ def to_parquet(path, df, compression=None, write_index=None, has_nulls=None,
                                            fixed_text=fixed_text,
                                            object_encoding=object_encoding)
 
+    if append:
+        pf = fastparquet.api.ParquetFile(path, open_with=myopen)
+        if pf.file_scheme != 'hive':
+            raise ValueError('Requested file scheme is hive, '
+                             'but existing file scheme is not.')
+
+        if len(set(pf.columns) ^ set(df._meta.columns)):
+            raise ValueError('Appended columns not the same.\n'
+                             'New: {} | Previous: {}'
+                             .format(pf.columns, list(df._meta.columns)))
+        else:
+            df = df[pf.columns]
+
+        fmd = pf.fmd
```

Comment:
Here we blew away the `fmd` generated from the data itself. Checks around the requested dtypes should go here: an exception should be raised not only if column names mis-match, but also if the inferred schemas do.